request_handlers.h

```diff
@@ -109,6 +109,22 @@ class NotFoundRequestHandler : public RequestHandler {
 };
 REGISTER_REQUEST_HANDLER(NotFoundRequestHandler);
 
+// Handle proxy server requests
+class ProxyRequestHandler : public RequestHandler {
```

Comment:
Could we call this `ReverseProxyRequestHandler` instead? That way, we can differentiate this vs. a potential bonus feature of (forward) proxy functionality. 😸